src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -122,6 +124,50 @@ function matrix(A::Array{AbstractAlgebra.Generic.FreeModuleElem{T},1}) where T <
    return X
 end
 
+"""
+    upper_triangular_matrix(L::AbstractVector{T}) where T <: RingElem
+
+Return the upper triangular matrix whose non-zero entries (ordered lexicographically) are the elements of the sequence `L`.
```

Comment:
This text doesn't quite make sense if L contains zeroes.
```suggestion
Return the upper triangular matrix whose entries on and above the diagonal are the elements of the sequence `L`. 
```

Reply:
Do I remove the `(ordered lexicographically)` ?

Reply:
Yes; and also the "non-zero" is misleading, as L could contain zeros. So you could just take my complete suggestion.